tests/unit/extensions/test_expressjs.py

```diff
@@ -0,0 +1,217 @@
+# -*- Mode:Python; indent-tabs-mode:nil; tab-width:4 -*-
+#
+# Copyright 2024 Canonical Ltd.
+#
+# This program is free software: you can redistribute it and/or modify
+# it under the terms of the GNU General Public License version 3 as
+# published by the Free Software Foundation.
+#
+# This program is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU General Public License for more details.
+#
+# You should have received a copy of the GNU General Public License
+# along with this program.  If not, see <http://www.gnu.org/licenses/>.
+import pytest
+from rockcraft import extensions
+from rockcraft.errors import ExtensionError
+
+
+@pytest.fixture(name="expressjs_input_yaml")
+def expressjs_input_yaml_fixture():
+    return {
+        "name": "foo-bar",
+        "base": "ubuntu@24.04",
+        "build-base": "ubuntu@24.04",
+        "platforms": {"amd64": {}},
+        "extensions": ["expressjs-framework"],
+    }
+
+
+@pytest.fixture
+def expressjs_extension(mock_extensions, monkeypatch):
+    monkeypatch.setenv("ROCKCRAFT_ENABLE_EXPERIMENTAL_EXTENSIONS", "1")
+    extensions.register("expressjs-framework", extensions.ExpressJSFramework)
+
+
+@pytest.fixture
+def expressjs_project_name():
+    return "test-expressjs-project"
+
+
+@pytest.fixture
+def app_path(tmp_path):
+    app_path = tmp_path / "app"
+    app_path.mkdir(parents=True, exist_ok=True)
+    return app_path
+
+
+@pytest.fixture
+def package_json_file(app_path, expressjs_project_name):
+    (app_path / "package.json").write_text(
+        f"""{{
+    "name": "{expressjs_project_name}",
+    "scripts": {{
+        "start": "node ./bin/www"
+    }}
+}}"""
+    )
+
+
+@pytest.mark.usefixtures("expressjs_extension", "package_json_file")
+def test_expressjs_extension_default(
```

Comment:
let's put the 4 scenarios here:

* 24.04 base and node not customised
* 24.04 base and node customised
* bare base and node not customised
* bare base and node version customised

Then we don't need the sub functions. It would probably be best to do this using parametrize